core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -119,8 +121,8 @@ public final Object visitGetKeysInGroupCommand(InvocationContext ctx, GetKeysInG
          return invokeNext(ctx, command);
       }
       Address primaryOwner = dm.getCacheTopology().getDistribution(command.getGroupName()).primary();
-      CompletableFuture<Map<Address, Response>> future = rpcManager.invokeRemotelyAsync(
-            Collections.singleton(primaryOwner), command, defaultSyncOptions);
+      CompletionStage<Map<Address, Response>> future = rpcManager
+            .invokeCommand(primaryOwner, command, new MapResponseCollector(false, 1), rpcManager.getSyncRpcOptions());
```

Comment:
Actually I wanted to use `SingleResponseCollector` for all the RPCs with a single target, but I only changed a few so far. Of course, I don't want to use that for long either: ideally I'd like to use custom `ResponseCollector` implementations everywhere.